File: Java/Language/Basics/src/SalesPerson.java
package payroll;

//subclassing
public class SalesPerson extends Employee {

	private double sales;

	public SalesPerson(int h, float r, double s) {
		super(h, r); //calling superclass (Employee) constructor
		sales = s;
	}

	public double getSales() { return sales; }

	//overriding superclass method
	public double income() {
		double amount = super.income(); // calling method from superclass
		if(sales >= 10000)
			amount += 0.05 * sales;
		return amount;
	}
}

File: CSharp/Basics/ClassTypeTest/Payroll/SalesPerson.cs
namespace Payroll
{
	public class SalesPerson : Employee
	{
		public double Sales { get; set; }

		public SalesPerson(int hours, float rate, double sales) : base(hours, rate)
		{
			Sales = sales;
		}

		public override double GetIncome()
		{
			double commission = Sales >= 20000 ? 0.05 * Sales : 0;
			return base.GetIncome() + commission;
		}
	}
}

